pollination-apps/app-developers/integrating-the-pollination-viewer.md

```diff
@@ -0,0 +1,131 @@
+
+The Streamlit component can be installed like this:
+```shell
+$ pip install pollination-streamlit-viewer
+```
+
+This guide will focus on the Streamlit component because that what will be most useful to developers of Pollination Apps. If you're interested in learning more about the React component, please reach out through our forums.
+
+The first resource you should know about is the sample viewer app: </br>
+https://app.pollination.cloud/pollination/apps/viewer-sample-app/?tab=live+app
+
+The code for this app is located here: </br>
+https://github.com/pollination-apps/pollination-viewer-sample/blob/main/app.py
+
+This app exposes all the configurations and features of pollination-streamlit-viewer.
+
+Here's a screenshot of the viewer itself:
+![](../../.gitbook/assets/integrating-the-pollination-viewer/pollination-viewer-model-loaded.png)
+
+
+### Instantiating the `viewer` in Streamlit
+
+```python
+def viewer(
+     key: str, *,
+     content: bytes = None,
+     toolbar: bool = True,
+     sidebar: bool = True,
+     subscribe: bool = False,
+     clear: bool = True,
+     action_stack: list = [],
+     style: Dict = None)
+```
+| Argument | Description |
+|----------|-------------|
+| key | A unique string for each instance of the viewer. |
+| content | A `.vtkjs` file (see the section below Loading a File) |
+| toolbar | A `boolean` that toggles the toolbar visibility. Default is `True`. |
+| sidebar | A boolean that toggles the sidebar visiblity. Default is `True`. |
+| subscribe | A boolean that toggles subscription to the VTKJS camera and renderer content. Default is `False`. |
+| clear | A boolean to clear the current contents from the viewer before loading new content. Default is `True`. |
+| action_stack | The action stack is an advanced feature that allows the streamlit component to send actions to the React component's dispatch function. A set of useful actions are demonstrated in the sample app. |
+| style | A dictionary to set the style for the viewer. The key and values can be any CSS style attribute. Default is:  `{"border": "1px solid #d0d7de", "borderRadius": "2px"}` |
+
+### Return Value
+The value returned by `viewer` represents the scene that has been loaded into the viewer. This object is represented as a deeply nested dictionary with fields that often follow the structure of vtk.js objects. By default, this state will contain a single value called scene with an array of all the actors in the scene. This default return value will not update as you update the scene using the viewer controls. This is useful for reading default values, and `actor[id]`s. Those `actor[id]`s are used by some actions to modify actor properties.
+
+If you want to read and respond to values as the viewer state changes, use the `subscribe` toggle described below.
+
+### Loading a File
+You can make use of the Streamlit provided st.file_uploader component to load a .vtkjs file.
+```python
```

Comment:
I might be wrong but you usually want to leave a blank line between the lines in a markdown document. Otherwise, they may not be rendered correctly on some platforms.